### PR TITLE
fix(useStorage): fixed data desynchronization in case of clearing storage

### PR DESCRIPTION
**Pull request by yuryrudey**
### Description

Fixes data desynchronization in case of clearing storage(using `storage.clear()` or manually)
Condition to detect storage clearing is taken from [documentation](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event#event_properties)(pay attention to description of `key` property)

### Additional context

Using `useStorage` I noticed that after clearing storage in Chrome DevTools, my computed property wasn't recomputed. The cause was condition in the storage event handler that ignored this case

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useStorage/index.ts [hunk 1]
```diff
@@ -218,7 +218,15 @@ export function useStorage<T extends(string | number | boolean | object | null)>
   }
 
   function update(event?: StorageEvent) {
-    if (event && event.key !== key)
+    if (event?.storageArea !== storage)
+      return
```
Comment (yuryrudey) [inline packages/core/useStorage/index.ts:222]: This condition also fixes bug with using `useStorage` on `sessionStorage` and `localStorage` with the same key

File: packages/core/useStorage/index.ts [hunk 2]
```diff
@@ -218,7 +218,15 @@ export function useStorage<T extends(string | number | boolean | object | null)>
...
+
+    if (event?.key === null) {
```
Comment (yuryrudey) [inline packages/core/useStorage/index.ts:224]: According to documentation, when `key` is `null`, it means that storage has been cleared

File: packages/core/useStorage/index.ts [hunk 3]
```diff
@@ -218,6 +215,14 @@ export function useStorage<T extends(string | number | boolean | object | null)>
   }
 
   function update(event?: StorageEvent) {
+    if (event && event.storageArea !== storage)
+      return
```
Comment (yuryrudey) [inline packages/core/useStorage/index.ts:219]: This condition also fixes overlapping bug when using `useStorage` on `sessionStorage` and `localStorage` with the same key

File: packages/core/useStorage/index.ts [hunk 4]
```diff
@@ -218,6 +215,14 @@ export function useStorage<T extends(string | number | boolean | object | null)>
...
+
+    if (event && event.key === null) {
+      data.value = rawInit
+      return
+    }
```
Comment (yuryrudey) [inline packages/core/useStorage/index.ts:224]: According to documentation, when key is null, it means that storage has been cleared

**Review by antfu — APPROVED** [review]

**Pull request merged**